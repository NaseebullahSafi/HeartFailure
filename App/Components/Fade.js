// predefined components from react
import React, { Component } from 'react'

// animated library from react to animate components
import {Animated} from 'react-native';

/**
 * This component will give fading animation to components
 */
export default class Fade extends Component {
  constructor(props) {
    super(props);
    this.state = {
      visible: props.visible,
    };
  };

  /**
   * When the component has been mounted, give a default value
   * for the animation initial state
   * ==============================================================
   */
  componentWillMount() {
    this._visibility = new Animated.Value(this.props.visible ? 1 : 0);
  }

  /**
   * When the component gets new props, start the animation, and
   * when it is done, tell the component that the animation has
   * finished
   * ==============================================================
   * @param nextProps
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.visible) this.setState({ visible: true });

    // start the animation with the visibility
    Animated.timing(this._visibility, {
      toValue: nextProps.visible ? 1 : 0,
      duration: 300,
    }).start(() => {
      this.setState({ visible: nextProps.visible });
    });
  }

  /**
   * Show the component when it is ready
   * ==============================================================
   * @return {XML}
   */
  render() {
    const {style, children, ...rest } = this.props;

    // let us define the fading props here by giving opacity
    const containerStyle = {
      opacity: this._visibility.interpolate({
        inputRange: [0, 1], outputRange: [0, 1],
      }),
      transform: [{
        scale: this._visibility.interpolate({
          inputRange: [0, 1], outputRange: [1.1, 1]
        })
      }],
    };

    // merge them together
    const combinedStyle = [containerStyle, style];

    // return it
    return (
      <Animated.View style={this.state.visible ? combinedStyle : containerStyle} {...rest}>
        {this.state.visible ? children : null}
      </Animated.View>
    );
  }
}
