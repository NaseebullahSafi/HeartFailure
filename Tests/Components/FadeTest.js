// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// the component that will be tested
import Fade from "../../App/Components/Fade";

Enzyme.configure({ adapter: new Adapter() });

describe('Testing <Fade /> component', () => {
  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <Fade visible={false}/>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
});

