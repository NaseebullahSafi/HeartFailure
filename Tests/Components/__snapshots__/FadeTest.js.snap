// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Fade /> component renders as expected 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 0,
      "transform": Array [
        Object {
          "scale": 1.1,
        },
      ],
    }
  }
  visible={false}
/>
`;
